### PR TITLE
fix (#minor); dforce; fixed QA issues reported in #868

### PR DESCRIPTION
**Pull request by tnkrxyz**
Fixed QA issues reported in #868
1. Fixed underlying token conflict: ZERO_ADDRESS is ETH on mainnet, BNB on BSC
2. Fixed protocol.mintedTokens and protocol.mintedTokenSupplies
3. Fixed financialDailySnapshot.mintedTokenSupplies (compound-forks/src/mapping.ts)
4. ARBITRUM_BLOCKS_PER_YEAR and ARBITRUM_BLOCKS_PER_DAY (also in #1049; will rebase this PR if #1049 is merged first)
5. bumped subgraph version to 1.2.0

### CODE REVIEW

**Comment by tnkrxyz** [comment]
See also #1106

**Comment by tnkrxyz** [comment]
@dmelotik , @0xbe1 , or @this-username-is-taken , can one of you help review? thx!

File: deployment/deployment.json
Comment (dmelotik) [inline deployment/deployment.json:1189]: A patch should only update the last number: https://semver.org/#summary

File: subgraphs/compound-forks/protocols/dforce/src/mapping.ts
Comment (dmelotik) [inline subgraphs/compound-forks/protocols/dforce/src/mapping.ts:637]: Be careful with ordering here and checking that way (https://github.com/messari/subgraphs/blob/master/docs/Mapping.md#array-sorting-when-querying)

**Review by dmelotik — COMMENTED** [review]
Nice work! A couple things, I now think we are enforcing rebasing onto the latest master, so you will have to do that.

I see dForce does have minted tokens, are we including this in the TVL on our subgraph?

And @ishraq8 for a second opinion. dForce minted tokens seem to be apart of their synthetics offerings, should this be included in this subgraph? To me it seems out of scope.

**Comment by ishraq8** [comment]
@dmelotik The synthetics in dforce seems to work somewhat similarly to synthetix (CDP and then goes to trade synthetic assets). The dforce synthetics while it can use the similar depositm, the mechanics are separate from the borrowing and may probably need it's own subgraph in the future like a CDP. Their synthetics market is ~$400K

**Comment by dmelotik** [comment]
@this-username-is-taken for visibility, do we remove this from the dforce subgraph then?

File: deployment/deployment.json
Comment (tnkrxyz) [reply deployment/deployment.json:1189]: Maybe I should reword the title of the PR instead. I feel the changes are substantial enough to warrant a minor version increase.

File: subgraphs/compound-forks/protocols/dforce/src/mapping.ts
Comment (tnkrxyz) [reply subgraphs/compound-forks/protocols/dforce/src/mapping.ts:637]: The purpose of the whole chunk of code (L625-649) is to make sure the order of mintedToken and mintedTokenSupplies arrays are consistent.

**Comment by tnkrxyz** [comment]
> Nice work! A couple things, I now think we are enforcing rebasing onto the latest master, so you will have to do that.
I believe I did that; I will rebase again before merging.

> 
> I see dForce does have minted tokens, are we including this in the TVL on our subgraph?
They are not included. Since (most) minted tokens are borrowed against collateral, including them would double count? 

> 
> And @ishraq8 for a second opinion. dForce minted tokens seem to be apart of their synthetics offerings, should this be included in this subgraph? To me it seems out of scope.

@this-username-is-taken and I talked a while back when I first started working on the dforce subgraph and he thought synthetics should not be included. Welcome second/third opinion here.

**Comment by ishraq8** [comment]
Thanks @tnkrxyz! Looks like there is consensus that the synthetics component should be handled separately.

**Comment by dmelotik** [comment]
> I believe I did that; I will rebase again before merging.

I was looking at this, but we will see if it is an issue when we are actually ready to merge: 
Screen Shot 2022-10-04 at 10 13 31 AM

I guess my question is should we have mintedTokens in the dForce subgraph? I am kinda confused there. I think we could also merge this PR then in a separate PR remove the mintedTokens mappings. Up to you really.

File: deployment/deployment.json
Comment (dmelotik) [reply deployment/deployment.json:1189]: I misread the title, I thought it said #patch . I guess the way I think of it is that since nothing was added to the schema there were no functionality changes. But honestly I am not totally sure

**Comment by tnkrxyz** [comment]
> I was looking at this, but we will see if it is an issue when we are actually ready to merge: Screen Shot 2022-10-04 at 10 13 31 AM

Rebased again.

> I guess my question is should we have mintedTokens in the dForce subgraph? I am kinda confused there. I think we could also merge this PR then in a separate PR remove the mintedTokens mappings. Up to you really.

My understanding is that mintedTokens and mintedTokenSupplies are specific for protocols like dforce, inverse-finance, and markedao, that mints stable coins, so they should be included here. Ping @this-username-is-taken for confirmation that is still the case.

**Comment by this-username-is-taken** [comment]
That's right, we should include minted tokens (similar to makerdao, abra, etc) so there is a way to track them

File: deployment/deployment.json
Comment (dmelotik) [inline deployment/deployment.json:2558]: remove this update

**Comment by dmelotik** [comment]
Small update and I think we are ready to merge!

Note, same thing with the new linting checker. Ignore this for now and we will make a separate PR to tackle that since there is a lot in it.

**Comment by tnkrxyz** [comment]
> Small update and I think we are ready to merge!
> 
> Note, same thing with the new linting checker. Ignore this for now and we will make a separate PR to tackle that since there is a lot in it.

Since #1157 is now merged, I'm rebasing to get the fix.

**Review by dmelotik — APPROVED** [review]

**Pull request merged**
